core/src/main/java/org/apache/accumulo/core/client/summary/summarizers/EntryLengthSummarizer.java

```diff
@@ -0,0 +1,329 @@
+
+        // VALUES
+        if (v.getSize() < minValue) {
+          minValue = v.getSize();
+        }
+
+        if (v.getSize() > maxValue) {
+          maxValue = v.getSize();
+        }
+
+        sumValues += v.getSize();
+
+        if (v.getSize() == 0) {
+          idx = 0;
+        } else {
+          idx = IntMath.log2(v.getSize(), RoundingMode.HALF_UP);
+        }
+
+        valueCounts[idx]++;
+
+        // TOTAL
+        total++;
+      }
+
+      @Override
+      public void summarize(StatisticConsumer sc) {
+        sc.accept(MIN_KEY_STAT, (minKey != Long.MAX_VALUE ? minKey:0));
+        sc.accept(MAX_KEY_STAT, (maxKey != Long.MIN_VALUE ? maxKey:0));
+        sc.accept(SUM_KEYS_STAT, sumKeys);
+
+        for (int i = 0;i < keyCounts.length;i++) {
+          if(keyCounts[i] > 0) {
+            sc.accept("key.logHist."+i, keyCounts[i]);
+          }
+        }
+
+        sc.accept(MIN_ROW_STAT, (minRow != Long.MAX_VALUE ? minRow:0));
+        sc.accept(MAX_ROW_STAT, (maxRow != Long.MIN_VALUE ? maxRow:0));
+        sc.accept(SUM_ROWS_STAT, sumRows);
+
+        for (int i = 0;i < rowCounts.length;i++) {
+          if(rowCounts[i] > 0) {
+            sc.accept("row.logHist."+i, rowCounts[i]);
+          }
+        }
+
+        sc.accept(MIN_FAMILY_STAT, (minFamily != Long.MAX_VALUE ? minFamily:0));
+        sc.accept(MAX_FAMILY_STAT, (maxFamily != Long.MIN_VALUE ? maxFamily:0));
+        sc.accept(SUM_FAMILIES_STAT, sumFamilies);
+
+        for (int i = 0;i < familyCounts.length;i++) {
+          if(familyCounts[i] > 0) {
+            sc.accept("family.logHist."+i, familyCounts[i]);
+          }
+        }
+
+        sc.accept(MIN_QUALIFIER_STAT, (minQualifier != Long.MAX_VALUE ? minQualifier:0));
+        sc.accept(MAX_QUALIFIER_STAT, (maxQualifier != Long.MIN_VALUE ? maxQualifier:0));
+        sc.accept(SUM_QUALIFIERS_STAT, sumQualifiers);
+
+        for (int i = 0;i < qualifierCounts.length;i++) {
+          if(qualifierCounts[i] > 0) {
+            sc.accept("qualifier.logHist."+i, qualifierCounts[i]);
+          }
+        }
+
+        sc.accept(MIN_VISIBILITY_STAT, (minVisibility != Long.MAX_VALUE ? minVisibility:0));
+        sc.accept(MAX_VISIBILITY_STAT, (maxVisibility != Long.MIN_VALUE ? maxVisibility:0));
+        sc.accept(SUM_VISIBILITIES_STAT, sumVisibilities);
+
+        for (int i = 0;i < visibilityCounts.length;i++) {
+          if(visibilityCounts[i] > 0) {
+            sc.accept("visibility.logHist."+i, visibilityCounts[i]);
+          }
+        }
+
+        sc.accept(MIN_VALUE_STAT, (minValue != Long.MAX_VALUE ? minValue:0));
+        sc.accept(MAX_VALUE_STAT, (maxValue != Long.MIN_VALUE ? maxValue:0));
+        sc.accept(SUM_VALUES_STAT, sumValues);
+
+        for (int i = 0;i < valueCounts.length;i++) {
+          if(valueCounts[i] > 0) {
+            sc.accept("value.logHist."+i, valueCounts[i]);
+          }
+        }
+
+        sc.accept(TOTAL_STAT, total);
+      }
+
+    };
+  }
+
```

Comment:
Could possibly a have static method like the following to cut down on redundant code.

```java
   private static merge(String prefix, Map<string, Long> stats1, Map<String, Long> stats2) {
        // merge min, max, sum, and hist for all props with prefix.
   }
```

This method could be called for each prefix.